CMakeLists.txt

```diff
@@ -1,84 +1,93 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(urg_node)
 
-find_package(catkin REQUIRED COMPONENTS
-  diagnostic_updater dynamic_reconfigure laser_proc message_generation
-  nodelet rosconsole roscpp sensor_msgs std_msgs std_srvs tf urg_c
-)
-set_directory_properties(PROPERTIES COMPILE_OPTIONS "-std=c++11")
+# Default to C++14
+if(NOT CMAKE_CXX_STANDARD)
+  set(CMAKE_CXX_STANDARD 14)
+endif()
+if(CMAKE_COMPILER_IS_GNUCXX OR CMAKE_CXX_COMPILER_ID MATCHES "Clang")
+  add_compile_options(-Wall -Wextra -Wpedantic)
+endif()
 
-# Dynamic reconfigure support
-generate_dynamic_reconfigure_options(cfg/URG.cfg)
+find_package(ament_cmake REQUIRED)
+find_package(Boost COMPONENTS thread REQUIRED)
+find_package(diagnostic_updater REQUIRED)
+find_package(laser_proc REQUIRED)
+find_package(rclcpp REQUIRED)
+find_package(sensor_msgs REQUIRED)
+find_package(std_srvs REQUIRED)
+find_package(tf2 REQUIRED)
+find_package(urg_c REQUIRED)
+find_package(urg_node_msgs REQUIRED)
 
-add_message_files(FILES
-  Status.msg
+add_library(urg_c_wrapper src/urg_c_wrapper.cpp)
+target_include_directories(urg_c_wrapper
+  PUBLIC
+  $<BUILD_INTERFACE:${CMAKE_CURRENT_SOURCE_DIR}/include>
+  $<INSTALL_INTERFACE:include>
+  ${Boost_INCLUDE_DIR}
 )
-
-generate_messages(
-  DEPENDENCIES std_msgs
+target_link_libraries(urg_c_wrapper ${Boost_LIBRARIES})
+ament_target_dependencies(urg_c_wrapper
+  rclcpp
+  sensor_msgs
+  urg_c
 )
 
-catkin_package(
-  INCLUDE_DIRS include
-  LIBRARIES urg_c_wrapper urg_node_driver
-  CATKIN_DEPENDS dynamic_reconfigure laser_proc message_runtime nodelet rosconsole
-  roscpp sensor_msgs std_msgs std_srvs urg_c
-  DEPENDS
+add_library(urg_node_driver SHARED src/urg_node_driver.cpp)
```

Comment:
This wouldn't link for me; I had to remove the `SHARED` from this line.  I suspect it's because the `urg_c_wrapper` library above is not marked SHARED, and they need to agree to be linked together.  Suggest removing `SHARED` from this line.

Reply:
that sounds strange. Why wouldn't you be able to link a static library to a shared one?

Reply:
Probably because the shared one is built with `-fPIC` and the static one without it.  Does it build for you?

Reply:
yup, builds cleanly for me - must be because of OSX being the better OS ;-)

Reply:
Re-iterating over this, I didn't see the point of having the two files compiled into its individual libraries, so I've put them together.
In order to make them components, they have to be a shared lib in order for poco to load that thing during runtime. Please let me know if that works for you.

Reply:
Sort of.  I now had to add the SHARED keyword to both the laser_proc and the urg_c stuff; with all of that in place, it now works.  I'll open additional PRs to those two to show you what I mean.